simulacrum/dataset.py

```diff
@@ -102,6 +104,42 @@ def zip_data(self, ty, length):
             res.append(f.name())
         return pd.Series(res)
 
+    @staticmethod
+    def uuid_data(ty, length):
+        """
+        Generate a column of random uuids.
+
+        :param length: The number of uuids.
+        :type length: int.
+        :return: The column of uuids.
+        :rtype: pd.Series
+
+        """
+        return pd.Series(list(map(lambda _: uuid4(), range(length))))
+
+    @staticmethod
+    def faker_data(ty, length):
+        """
+        Generate a column based on any faker data type.
+
+        :param ty: A configuration for the faker data. Must contain faker provider and related args as dict.
+        :param length: The number of rows wanted.
+        :param ty: dict.
+        :param length: The number of rows wanted.
+        :type length: int.
+        :return: The column of Faker data.
+        :rtype: pd.Series
+
+        """
+        try:
+            provider = ty["provider"]
+            del ty["provider"]
```

Comment:
oh ok I forgot that the "type" property is propagated through the related methods.

````python
for col, typ in coltypes_items:
    series_res[col] = ops[typ['type']](typ, length)
````   
I could be interesting to avoid to propagate useless properties through methods, instead of handling these properties inside these methods. Example:

````python
for col, typ in coltypes_items:
    data_builder = ops[typ['type']]
    del typ['type']
    series_res[col] = data_builder(typ, length)
````   
It's easier to understand and to maintain.
Are you ok ?